### PR TITLE
Add Support for Multi-conformer and ELF10 WBOs

### PR DESCRIPTION
**Pull request by SimonBoothroyd**
## Description

This PR adds support to the OpenEye toolkit wrapper for computing WBOs from multiple conformers, and adds a convenient way to compute them from ELF10 conformers in a single call.

This PR partially addresses #792.

## TODOs

- [X] Tag issue being addressed
- [X] Add [tests](https://github.com/openforcefield/openff-toolkit/tree/master/openff/toolkit/tests)
- [X] Update docstrings/[documentation](https://github.com/openforcefield/openff-toolkit/tree/master/docs), if applicable
- [X] [Lint](https://open-forcefield-toolkit.readthedocs.io/en/latest/developing.html#style-guide) codebase
- [x] Update [changelog](https://github.com/openforcefield/openff-toolkit/blob/master/docs/releasehistory.rst)

## Status

- [x] Ready to go

### CODE REVIEW

**Review by j-wags — CHANGES_REQUESTED** [review]
These changes look technically good, but I think that the result of merging this would put `master` in an unacceptable state -- the difference in `molecule.assign_fractional_bond_orders` behavior with the AmberToolsToolkitWrapper (first conf only) vs. OpenEyeToolkitWrapper (unweighted average over all confs) is substantial. 

So, two concrete asks:

1) Previously, you'd mentioned not including AmberToolsToolkitWrapper in these changes because the conformer minimization issue is unresolved. But I think that problem is orthogonal to this one. Since having both ToolkitWrappers do all-conformer averaging is something we can implement now to prevent the problem above, I think we should add the averaging to `ATTKW.assign_fractional_bond_orders` in this PR as well. 

2) It's likely that the behavior of the method currently called `am1-wiberg-elf10` will change in the future -- maybe, as we find which parameters yield improved charges, the backend implementation will change to have stronger/weaker restraints during minimization, or a different RMS cutoff, or we'll use a fraction other than 2%. The point at which the method becomes "distinct" will be unclear, but at some point we'll have a method which does a substantially different thing than what it does now. This could have undesired effects on force field performance. So we need to reconcile how we want to have names like `am1-wiberg-elf10` refer both to a) code that we can experiment with while avoiding undue concerns about consistency, and b) precise strings that can be included in our next OFFXML and be consistently supported by all future versions of OFFTK.

The second point is hard to solve. Basically, either
* we make the awkward statements "all of the subsequent versions of this implementation are about the same, no need to worry about the charge method changing significantly between releases" and at the same time "this implementation of the charge method is deficient enough that we are investing time and energy into changing it"
* we need to specify _exactly_ what these charge methods _aspire_ to do, and then say that even if our implementation changes from toolkit version to toolkit version, that we're inching ever closer to correctly implementing the aspirational behavior (this may be difficult -- how _exactly_ do we aspire to generate canonical conformers? -- but maybe some lesser level of detail is good enough?)
* we start adding version numbers to everything (eg. `am1-wiberg-elf10-offtk-0.9.1`), which will become a snowball of legacy support

I lean most toward the second option (specify-and-aspire) but am interested to hear your thoughts.

**Comment by j-wags** [comment]
Only the first one is an immediate blocker. 

On the second, my inclination is that we should identify a "aspirational implementation" -- Maybe "whatever ELF10 OE is doing in the 2020 toolkits". But we should write that down somewhere definitive (like the SMIRNOFF spec) by the time this method gets into a released force field.

**Comment by SimonBoothroyd** [comment]
> Previously, you'd mentioned not including AmberToolsToolkitWrapper in these changes because the conformer minimization issue is unresolved. But I think that problem is orthogonal to this one. Since having both ToolkitWrappers do all-conformer averaging is something we can implement now to prevent the problem above, I think we should add the averaging to ATTKW.assign_fractional_bond_orders in this PR as well.

My preference here would have been to just raise a `NotImplementedError` on the AT side when multiple conformers are provided given that the behaviour is not 100% understood. 

My thinking with the conformer minimisation is that it may lead to multiple of a users conformers being collapsed into a single conformer essentially depending on how well the minimiser performs, and thus leading to different behaviour than the user may expect. In practice this likely won't happen (with the exception of strong electrostatic interactions perhaps), but there is a possibility I guess.

I've made the change as requested however in 3f2a37e, with the disclaimer that I wouldn't recommend people use it just yet.

File: docs/releasehistory.rst [hunk 1]
```diff
@@ -34,6 +36,10 @@ Behavior changed
   factor for electrostatic interactions is now properly set by the value specified in the force
   field. Previously it fell back to a default value of 0.83333. The toolkit may now produce
   slightly different energies as a result of this change.
+- `PR #839 <https://github.com/openforcefield/openforcefield/pull/839>`_: The average WBO will now be returned when
+  multiple conformers are provided to ``assign_fractional_bond_orders`` using ``use_conformers`` and an OpenEye
+  toolkit wrapper is used. The behaviour when computing WBOs using the AmberTools toolkit wrapper is currently
+  unchanged.
```
Comment (j-wags) [inline docs/releasehistory.rst:42]:
```suggestion
  multiple conformers are provided to ``assign_fractional_bond_orders`` using ``use_conformers``.
```

**Review by j-wags — APPROVED** [review]
Thanks. I agree that it's a mixed bag, but given how stop-and-start development can be around here, I think it's best to keep the toolchains very in sync (otherwise I have a bad habit of forgetting these sorts of details until something really embarrassing happens)

**Pull request merged**